fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java

```diff
@@ -109,9 +109,22 @@ public BestIndexInfo selectBestMV(ScanNode scanNode) throws UserException {
         long start = System.currentTimeMillis();
         Preconditions.checkState(scanNode instanceof OlapScanNode);
         OlapScanNode olapScanNode = (OlapScanNode) scanNode;
+
+        ConnectContext connectContext = ConnectContext.get();
+        if (connectContext != null && connectContext.getSessionVariable().isUseV2Rollup()) {
+            // if user set `use_v2_rollup` variable to true, and there is a segment v2 rollup,
+            // just return the segment v2 rollup, because user want to check the v2 format data.
+            OlapTable tbl = olapScanNode.getOlapTable();
+            String v2RollupIndexName = MaterializedViewHandler.NEW_STORAGE_FORMAT_INDEX_NAME_PREFIX + tbl.getName();
+            Long v2RollupIndexId = tbl.getIndexIdByName(v2RollupIndexName);
+            if (v2RollupIndexId != null) {
+                return new BestIndexInfo(v2RollupIndexId, false, "use_v2_rollup is true");
+            }
+        }
+
         Map<Long, List<Column>> candidateIndexIdToSchema = predicates(olapScanNode);
         long bestIndexId = priorities(olapScanNode, candidateIndexIdToSchema);
```

Comment:
Maybe We should check the candidate index firstly and choose the best rollup in priorities.